### PR TITLE
Oophw

### PR DESCRIPTION
**Pull request by Samotkan1998**

### CODE REVIEW

File: src/functions/Functions.php [hunk 1]
```diff
@@ -93,13 +91,6 @@ public function countArguments(): array
      */
     public function countArgumentsWrapper(): array
     {
-        $arg = func_get_args();
-        foreach ($arg as $value) {
-            if (!is_string($value)) {
```
Comment (yderevianko-cartera) [inline src/functions/Functions.php:98]: why did you delete this code?

File: src/oop/app/src/Models/Movie.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace src\oop\app\src\Models;
+
+class Movie implements MovieInterface
+{
+    private $title;
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Models/Movie.php]: you can use php 7 string type hint

File: src/oop/app/src/Parsers/FilmixParserStrategy.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        preg_match('%h1 class="name" itemprop="name">(.*?)</h1>%', $siteContent, $matches);
+        $title = iconv('windows-1251', 'utf-8', $matches[1]);
+
+        preg_match('%<a class="fancybox".*href="([^"]*)"%', $siteContent, $matches);
+        $poster = $matches[1];
+
+        preg_match('%<div class="full-story">(.*?)<\/div>%', $siteContent, $matches);
+        $description = iconv('windows-1251', 'utf-8', $matches[1]);
+
+        return new Movie($title, $poster, $description);
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/FilmixParserStrategy.php]: try to inject Movie instead of craeting new instance

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+     */
+    public function parseContent(string $siteContent): Movie
+    {
+        $crawler = new Crawler($siteContent);
+
+        $title = $crawler->filter('.ftitle > h1')->text();
+        $poster = $crawler->filter('.fposter > a')->attr('href');
+        $description = $crawler->filter('.fdesc')->text();
+
+        return new Movie($title, $poster, $description);
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: try to inject Movie instead of craeting new instance

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+class KinoukrDomCrawlerParserAdapter implements ParserInterface
+{
+    /**
+     * @param string $siteContent
+     * @return Movie
+     */
+    public function parseContent(string $siteContent): Movie
+    {
+        $crawler = new Crawler($siteContent);
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: try to inject Crawler instead of craeting new instance

File: src/oop/app/src/Scrapper.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+{
+    private $transporter;
+    private $parser;
+
+    public function __construct(TransportInterface $transporter, ParserInterface $parser)
+    {
+        $this->transporter = $transporter;
+        $this->parser = $parser;
+    }
+    public function getMovie(string $url): Movie
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Scrapper.php:29]: please add empty line

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/functions/Functions.php (hunk 1 above)
Comment (Samotkan1998) [reply src/functions/Functions.php:98]: Original repo has been changed and those lines were removed.

**Review by yderevianko-cartera — APPROVED** [review]
